src/Php82/Php82.php

```diff
@@ -174,6 +174,42 @@ public static function ini_parse_quantity(string $value): int
 
                 return 0;
             }
+
+            $digits_consumed = $digits;
+            /* Ignore leading whitespace. */
+            while ($digits_consumed < $str_end && false !== strpos($ctype_space, $value[$digits_consumed])) {
+                ++$digits_consumed;
+            }
+            if ($digits_consumed !== $str_end && ($value[$digits_consumed] === '+' || $value[$digits_consumed] === '-')) {
+                ++$digits_consumed;
+            }
+
+            if ($value[$digits_consumed] === '0') {
+                /* Value is just 0 */
+                if ($digits_consumed + 1 === $str_end) {
+                    return 0;
```

Comment:
I believe `goto evaluation;` or actually adding the same helper function `zend_ini_consume_quantity_prefix` would match the “style” of the original implementation better.

But trying to understand the code, I realize that the original implementation is needlessly complicated / still allows for funky stuff like `0x+0` and `0x     0`. /cc @girgias

Reply:
I don't think it allows a second + or whitespace, this is why we needed to use two loops to handle the prefix and suffix shenanigans. But I might have implemented this wrong, I do know there are some weird cases that are still possible tho :-/

Reply:
> goto evaluation

updated!